app/src/main/kotlin/com/simplemobiletools/calculator/helpers/MoneyCalculatorImpl.kt

```diff
@@ -91,13 +92,22 @@ class MoneyCalculatorImpl(calculator: Calculator,taxCalculator: TaxCalculator, v
         if(false){
 
         }else{
-            tCallback!!.spawnTaxModal()
+            moneyCallback!!.spawnTaxModal()
         }
     }
 
-    fun performTaxing(location:String){
+    fun calculateCurrencyConversion() {
```

Comment:
Why do you need to do this it's very unorthodox to have the controller trigger ui events. Please explain.

Reply:
I agree. @todd829 intends to correct this in a refactoring.